full.async/src/full/async.clj

```diff
@@ -133,27 +133,28 @@
   nil, puts it in the results channel. Returns the results channel. Closes the
   returned channel when the input channel has been completely consumed and all
   objects have been processed."
-  [f> parallelism ch]
-  {:pre [(fn? f>)
-         (and (integer? parallelism) (pos? parallelism))
-         (instance? ReadPort ch)]}
-  (let [results (async/chan)
-        threads (atom parallelism)]
-    (dotimes [_ parallelism]
-      (go
-        (loop []
-          (when-let [obj (<! ch)]
-            (if (instance? Throwable obj)
-              (do
-                (>! results obj)
-                (async/close! results))
-              (do
-                (when-let [result (<! (f> obj))]
-                  (>! results result))
-                (recur)))))
-        (when (zero? (swap! threads dec))
-          (async/close! results))))
-    results))
+  ([f> results parallelism ch]
```

Comment:
Ok I guess it would make it less convenient with ->>. Can you maybe still move it as 3rd not second arg and perhaps rename args to something more meaningful: [f> parallelism out-ch in-ch]